src/providers/RemoteAttributesCompletionProvider.ts

```diff
@@ -0,0 +1,69 @@
+import fetch from 'node-fetch';
+import * as vscode from 'vscode';
+
+export class RemoteAttributesCompletionProvider implements vscode.CompletionItemProvider {
+
+    private hasBeenInitialized = false;
+    private cachedAttributes: AttributeDefinition[] = [];
+
+    async provideCompletionItems(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken, context: vscode.CompletionContext): vscode.ProviderResult<vscode.CompletionList<vscode.CompletionItem> | vscode.CompletionItem[]> {
+        if(!this.hasBeenInitialized) {
+            await this.initCache();
+        }
+        
+        const completionItems = [];
+        for (const attributeDefinition of this.cachedAttributes) {
+            // TODO: handle range
+            const completionItem: vscode.CompletionItem = new vscode.CompletionItem(attributeDefinition.name);
+            completionItem.insertText = `{${attributeDefinition.name}}`;
+            completionItem.detail = attributeDefinition.defaultValue;
+            completionItem.filterText = `${attributeDefinition.name} ${attributeDefinition.defaultValue}`;
+            completionItem.documentation = `Attribute definition coming from ${attributeDefinition.origin}`;
```

Comment:
surely this will need to be improved so that end-users better know where to look at